src/Microsoft.Extensions.Configuration.AzureAppConfiguration/Models/KeyValueSelector.cs

```diff
@@ -52,9 +62,33 @@ public override bool Equals(object obj)
         /// <returns>A hash code for the current object.</returns>
         public override int GetHashCode()
         {
+            int tagsFilterHash = 3;
+
+            if (TagsFilter != null && TagsFilter.Any())
+            {
+                var sortedTags = new SortedSet<string>(TagsFilter);
+
+                if (sortedTags.Any())
```

Comment:
Why check for sortedTags.Any() here if its already been checked with TagsFilter.Any()

Reply:
Oh yea good point my fault